src/components/Modal/NewsletterSubscriptionModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react'
+import { Modal, ModalProps } from 'decentraland-ui/dist/components/Modal/Modal'
+import { Header } from 'decentraland-ui/dist/components/Header/Header'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { Field } from 'decentraland-ui/dist/components/Field/Field'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import Paragraph from 'decentraland-gatsby/dist/components/Text/Paragraph'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+
+import './ProposalModal.css'
+import './NewsletterSubscriptionModal.css'
+import Label from 'decentraland-gatsby/dist/components/Form/Label'
+import { Governance } from '../../api/Governance'
+import { NewsletterSubscriptionResult } from '../../entities/NewsletterSubscription/types'
+import useAsyncTask from 'decentraland-gatsby/dist/hooks/useAsyncTask'
+import useAuthContext from 'decentraland-gatsby/dist/context/Auth/useAuthContext'
+const check = require('../../images/icons/check-cloud.svg')
+
+export const NEWSLETTER_SUBSCRIPTION_KEY: string = 'org.decentraland.governance.newsletter_subscription'
+export const ANONYMOUS_USR_SUBSCRIPTION: string = 'anonymous_subscription'
+
+export type NewsletterSubscriptionModalProps = Omit<ModalProps, 'children'> & {
+  onSubscriptionSuccess?: () => void
+  subscribed: boolean
+}
+
+export function NewsletterSubscriptionModal({ onSubscriptionSuccess, subscribed, ...props }: NewsletterSubscriptionModalProps) {
+  const [account] = useAuthContext()
+  const l = useFormatMessage()
+  const [isValid, setIsValid] = useState(true);
+  const [message, setMessage] = useState('');
+  const [email, setEmail] = useState('')
+
+  const emailRegex = /\S+@\S+\.\S+/;
+
+  const validateEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const email = event.target.value;
+    if (emailRegex.test(email)) {
+      setIsValid(true);
+      setMessage('');
+      setEmail(email)
+    } else {
+      setIsValid(false);
+      setMessage(l('modal.newsletter_subscription.email_error_message') || '');
+      setEmail(email)
```

Comment:
if you are setting more than one variable at same time a composite state is preferred:

```ts
const [ state, setState ] = useState<{ isValid, message, email }>({ isValid: true, message: '', email: ''})

// ...

setState({
  isValid: true,
  message: '',
  email: email
})
```